scripts/check-dependencies.js

```diff
@@ -170,15 +170,43 @@ const check = () => {
     // Maps package-name -> the peerDependencies section for each monorepo-local
     // package, used for sense-checking dependencies later on.
     const peerDependenciesByPackage = {}
+    const engines = {}
+    let pkgVersion
+
     listPackages().forEach((pkgDir) => {
         const pkgFile = path.join(pkgDir, 'package.json')
         const pkg = readJSON(pkgFile)
+
         peerDependenciesByPackage[pkg.name] = pkg.peerDependencies || {}
+
+        if (!engines.node && pkg.engines) {
+            engines.node = pkg.engines.node
+        }
+        if (!engines.npm && pkg.engines) {
+            engines.npm = pkg.engines.npm
+        }
+
+        if (!pkgVersion) {
+            pkgVersion = pkg.version
+        }
```

Comment:
I think you're trying to set up variables that keep track of what the engine and version _should_ be across all packages here, to make sure they are consistent. I understand that. How will the code handle the case where the _first_ package you encounter has the wrong version and the wrong engines setting?

If your packages look like this:

```
package1 = {engines: {node: "nonsense", npm: "nonsense"}, version: "nonsense"} 
package2 = {engines: {node: "14.1.1", npm: "7.1.3"}, version: "1.2.3"}
```

Then I think by the time this loop terminates you have set the engines and version variables to "nonsense". Then, later on, you'll use that to check that other packages are consistent with it.